state/execution_test.go

```diff
@@ -0,0 +1,300 @@
+	txs2 := types.Accs2TxOutputs(et.accOut)
+
+	et.acc2State(et.accIn)
+	_, res = getOrMakeOutputs(et.state, nil, txs1)
+	assert.True(res.IsOK(), "getOrMakeOutputs: error when sending to existing account")
+
+	mapRes2, res := getOrMakeOutputs(et.state, nil, txs2)
+	assert.True(res.IsOK(), "getOrMakeOutputs: error when sending to new account")
+
+	//test the map results
+	_, map2ok := mapRes2[string(txs2[0].Address)]
+	assert.True(map2ok, "getOrMakeOutputs: account output does not contain new account map item")
+
+}
+
+func TestValidateInputsBasic(t *testing.T) {
+	assert := assert.New(t)
+	et := newExecTest()
+
+	//validate input basic
+	txs := types.Accs2TxInputs(1, et.accIn)
+	res := validateInputsBasic(txs)
+	assert.True(res.IsOK(), fmt.Sprintf("validateInputsBasic: expected no error on good tx input. Error: %v", res.Error()))
+
+	txs[0].Coins[0].Amount = 0
+	res = validateInputsBasic(txs)
+	assert.True(res.IsErr(), "validateInputsBasic: expected error on bad tx input")
+
+}
+
+func TestValidateInputsAdvanced(t *testing.T) {
+	assert := assert.New(t)
+	et := newExecTest()
+
+	//validate inputs advanced
+	txs := et.getTx(1, et.accIn, et.accOut)
+
+	et.acc2State(et.accIn, et.accOut)
+	accMap, res := getInputs(et.state, txs.Inputs)
+	assert.True(res.IsOK(), fmt.Sprintf("validateInputsAdvanced: error retrieving accMap. Error: %v", res.Error()))
+	signBytes := txs.SignBytes(et.chainID)
+
+	//test bad case, unsigned
+	totalCoins, res := validateInputsAdvanced(accMap, signBytes, txs.Inputs)
+	assert.True(res.IsErr(), "validateInputsAdvanced: expected an error on an unsigned tx input")
+
+	//test good case sgined
+	et.signTx(txs, et.accIn, et.accOut)
+	totalCoins, res = validateInputsAdvanced(accMap, signBytes, txs.Inputs)
+	assert.True(res.IsOK(), fmt.Sprintf("validateInputsAdvanced: expected no error on good tx input. Error: %v", res.Error()))
+	assert.True(totalCoins.IsEqual(txs.Inputs[0].Coins.Plus(txs.Inputs[1].Coins)), "ValidateInputsAdvanced: transaction total coins are not equal")
+}
+
+func TestValidateInputAdvanced(t *testing.T) {
+	assert := assert.New(t)
+	et := newExecTest()
+
+	//validate input advanced
+	txs := et.getTx(1, et.accIn, et.accOut)
+
+	et.acc2State(et.accIn, et.accOut)
+	signBytes := txs.SignBytes(et.chainID)
+
+	//unsigned case
+	res := validateInputAdvanced(&et.accIn.Account, signBytes, txs.Inputs[0])
+	assert.True(res.IsErr(), "validateInputAdvanced: expected error on tx input without signature")
+
+	//good signed case
+	et.signTx(txs, et.accIn, et.accOut)
+	res = validateInputAdvanced(&et.accIn.Account, signBytes, txs.Inputs[0])
+	assert.True(res.IsOK(), fmt.Sprintf("validateInputAdvanced: expected no error on good tx input. Error: %v", res.Error()))
+
+	//bad sequence case
+	et.accIn.Sequence = 2
+	et.signTx(txs, et.accIn, et.accOut)
+	res = validateInputAdvanced(&et.accIn.Account, signBytes, txs.Inputs[0])
+	assert.True(res.IsErr(), "validateInputAdvanced: expected error on tx input with bad sequence")
```

Comment:
would be cool if we could check the actual error here (ie. abci BadSequence) and below (abci InsufficientFunds)